Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -51,10 +94,53 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
+            var newArray = ProcessArray(array, index);
+            return newArray;
+        }
+
+        private static int[] ProcessArray(int[] array, int index)
+        {
+            var isValid = IsValidCase(array, index);
+
+            if (!isValid)
+            {
+                return array;
+            }
+
+            array = Remove(array, index);
             return array;
         }
 
+        private static bool IsValidCase(int[] array, int index)
+        {
+            if (array == null || index < 0 || index == array.Length || array.Length == 0 || index > array.Length)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887669)